### ISSUE TITLE
Fonts fallback

### ISSUE THREAD

**Issue by gpakosz** [asciinema/agg#31]
Hello,

Is there a way to control the font fallback?

I tried to use `--font-family 'Source Code Pro, Hack Nerd Font Mono` but `Arial Unicode MS` is picked

```
[2023-02-23T22:29:23Z WARN  usvg::text::shaper] Fallback from Source Code Pro to Arial Unicode MS.
```

This is when using the `resvg` renderer

**Comment by gpakosz** [comment]
It doesn't log much beside multiple lines per frame saying
```
[2023-03-03T14:13:37Z WARN  usvg::text::shaper] Fallback from Source Code Pro to Arial Unicode MS.
```

**Comment by gpakosz** [comment]
With `fontdue`

```
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] looking up font for family=Source Code Pro, weight=400, style=Normal
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] found font with id=ID(829)
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] looking up font for family=Source Code Pro, weight=700, style=Normal
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] found font with id=ID(810)
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] looking up font for family=Source Code Pro, weight=400, style=Italic
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] found font with id=ID(817)
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] looking up font for family=Source Code Pro, weight=700, style=Italic
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] found font with id=ID(822)
[2023-03-03T14:15:03Z DEBUG agg::renderer::fontdue] looking up font for family=Noto Emoji, weight=400, style=Normal
```

Not sure why it picks `Noto Emoji` here

**Comment by sickill** [comment]
Noto Emoji is used for rendering emojis, it's not a fallback for text font. If you have it installed on your system then it will always look it up for emoji, this is expected.

As for `--font-family` try without spaces after commas, so like this: `--font-family 'Source Code Pro,Hack Nerd Font Mono`

**Comment by sickill** [comment]
I just made a small fix, https://github.com/asciinema/agg/commit/24e2667591257f384cf42f800283140aec3acbb2, which makes it work with extra spaces like you tried. Until it's released just pass fonts without spaces after commas.

**Comment by gpakosz** [comment]
Without a space, I don't have a warning but I don't get the proper rendering for `U+279C`. This glyph is not in `Source Code Pro` but is in `Hack Nerd Font`. It seems it's not picked though

**Comment by sickill** [comment]
Ah, yeah. There's only one font used, the first one found on the system is used for all characters when using default `fontdue` renderer. Does it fallback correctly for `--renderer=resvg` ?

**Comment by gpakosz** [comment]
No. I used `--font-family 'Source Code Pro,Hack Nerd Font Mono' --renderer=resvg` and I'm getting
```
[2023-03-28T14:37:55Z WARN  usvg::text::shaper] Fallback from Source Code Pro to Arial Unicode MS.
```

**Comment by sickill** [comment]
I see. So usvg library must be doing this fallback internally. And I think I know why it doesn't see your Hack font. This actually explains a lot (including other bugs). Thanks for the info.

**Comment by sickill** [comment]
I've just released 1.4.0 with fixed font fallback behaviour. Please give it a try, I'd appreciate feedback on whether it solved the above issue. Also, the default renderer (fontdue) should render better than resvg now.

**Comment by sickill** [comment]
Excellent :tada:
